libtenzir/builtins/operators/from_load_read.cpp

```diff
@@ -183,20 +259,44 @@ class from_plugin final : public virtual operator_parser_plugin {
   auto parse_operator(parser_interface& p) const -> operator_ptr override {
     auto usage = "from <loader> <args>... [read <parser> <args>...]";
     auto docs = "https://docs.tenzir.com/next/operators/sources/from";
-    auto l_name = p.accept_shell_arg();
-    if (!l_name) {
+    auto q = until_keyword_parser{"read", p};
+    auto l_name = q.accept_shell_arg();
+    if (not l_name) {
       diagnostic::error("expected loader name")
-        .primary(p.current_span())
+        .primary(q.current_span())
         .usage(usage)
         .docs(docs)
         .throw_();
     }
-    auto l_plugin = plugins::find<loader_parser_plugin>(l_name->inner);
-    if (!l_plugin) {
-      throw_loader_not_found(*l_name);
+    auto loader = std::unique_ptr<plugin_loader>{};
+    if (auto split = l_name->inner.find("://"); split != std::string::npos) {
+      auto substr
+        = [](located<std::string> x, size_t pos,
+             size_t count = std::string::npos) -> located<std::string> {
+        auto sub = x.inner.substr(pos, count);
+        auto sub_source = location::unknown;
+        if (x.source && (x.source.end - x.source.begin) == x.inner.size()) {
+          sub_source.begin = x.source.begin + pos;
+          sub_source.end = sub_source.begin + sub.length();
+        }
+        return {std::move(sub), sub_source};
+      };
+      auto scheme = substr(*l_name, 0, split);
+      auto path = substr(*l_name, split + 3);
```

Comment:
Both CAF and Arrow have their own [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986)-compliant URI parser. It probably makes a lot of sense to use it so that we can correctly the default parser for URIs like `s3://foo/bar.json?endpoint_override=localhost:3000`.

Reply:
And soon we have one from libcurl as well. 😉